client/lib/domains/reducer.js

```diff
@@ -82,6 +83,16 @@ function reducer( state, payload ) {
 			} );
 
 		case UpgradesActionTypes.PRIMARY_DOMAIN_SET_COMPLETED:
+			return updateSiteState( state, siteId, {
```

Comment:
This is a lot of logic - while it could be a bit simplified, how about a different approach? We can pass any parameters we want along with the action. I think it should be possible to pass the old and new `domainName`s - that way, we wouldn't have to iterate over the whole array, just update those two entries. While this will add a bit of logic on the view, I feel it's a cleaner overall approach.